c7n/resources/mq.py

```diff
@@ -111,24 +111,24 @@ class VpcFilter(VpcFilter):
     def get_related_ids(self, resources):
         client = local_session(self.manager.session_factory).client('ec2')
         related_ids = set()
+        sg_ids = set()
         for r in resources:
-            try:
-                sg_ids = r.get('SecurityGroups', [])
-            except IndexError:
-                continue
-            vpc_id = self.sg_vpc_dict.get(sg_ids[0])
+            sg_id = r.get('SecurityGroups', [])[0]
```

Comment:
In the (unlikely) event that `r` has no key `'SecurityGroups'`, this will provide a zero-element list. The `[0]` will then raise an `IndexError` exception that's not handled. I'm not sure this is a good design. A try:/except: block may handle this more cleanly.